pkg/plugins/golang/v2/scaffolds/internal/templates/makefile.go

```diff
@@ -67,60 +67,75 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
-all: manager
+all: build
 
-# Run tests
-test: generate fmt vet manifests
+##@ General
```

Comment:
So far all changes in the Makefile has been addressing only for v3. The rule/agreement only addresses bug fix to v2. 

The reason for that I because it might be a breaking change. For SDK tool for example, as a consumer, it would be since it has plugins ship customs on top.

Reply:
What do you mean by customs on top? They were just reorder in order to group them by categories.

Reply:
SDK changes the Makefile. Add things on that. 
It adds the help and changes the order and etc .. 

And then, we have been asking for this kind of the change just be done in v3 so far. Indeed when the help feature was discussed we decided that it should be done only for v3 as well. As we have a previous agreement of mainly only push bug fix to v2. So, I do not think that we ought to do that.

Reply:
The agreement was {N-2} bug fixes, {N-1} bug fixes and easy to back port features and enhancements, N all features. And go/v2 is still N or {N-1} (depending if you consider go/v3 to be N as it still is not implemented in any stable release). Either way, it is not {N-2} in any way so it is not restricted to bug fixes only.

About the paart that SDK adds helps and commands, I don't think we should not provide help ourselves just because SDK provides it, and you can still add new commands to the file, just add them in the correct category (or create a new one for SDK related targets) and add help messages so that the help target prints them out too.

Reply:
It is a change in the scaffold and is a breaking change for SDK(consumers). It will break the "plugins" that does custom on the Makefile, the logic to generate the samples, the tests and docs. 

Note that usually, a plugin to add features to a Kubebuilder project or a consumer will add targets in the Makefile or custom them. So, changes in its scaffold have a good impact. 

Regards SDK the burn will be the same because its using go/v3 already. In this way, we need to do changes in many places anyway. Also, the following change is not required to add help for example: 

````
# Backwards compatibility
manager: build ## Build manager binary (deprecated, use build target instead).
```

So, if we are ok changing the Makefile for v2 has any reason for v2 and v3 does not do exactly the same scaffold?

Reply:
Consumers/plugins do customs on the scaffold files. Usually, the Makefile will require new targets and etc.. So, changes in this file have a good chance to bring impact. 

Regarding specifically the effort to fix the breaking changes on SDK side, in this case, does not matter too much if the changes are applied only for v2 or v2 and v3. SDK has been using go/v3 already. So, it will probably affect anyway the "plugins" which does custom on the Makefile, the logic to generate the samples, the tests and docs. (for v2 and v3) 

Also, this PR does not only adds help or improve its text as described. For example,  the following change is not required to add help for example: 

```
# Backwards compatibility
manager: build ## Build manager binary (deprecated, use build target instead).
```

However, if we are ok in changing the Makefile for v2, has any reason for v2 and v3 does not do exactly the same scaffold less the test target using the controller-runtime?  It might reduce a few complexities.

Reply:
About the differences from v2/v3 there are 4:
1. v3 added the `preserveUnkownFields=false` option to controller-gen CRD generation. I think this is only needed for API and Webhook versions `v1` so that is the reason v2 doesn't have it as it is only supported in v3.
2. v3 the test target was changed to use controller-runtime's installation process.
3. v2 still has a `manager: build` target that acts as an alias so that if someone was using `make manager` they can still use it. For example they may have some automatic process that runs it and would break it if we don't keep the backwards compatibility.
4. `controller-gen` and `kustomize` installation scripts were modified for v3.

The rest of the files are now completely identical. They had more differences before.

About adding new targets, SDK could do two things:
1. Append `##@ SDK` as the end and add its targets under this new category.
2. Use the currently existing categories by finding `##@ <CATEGORY>` and add its targets below.

About the P.S.: I did addd that info in the first message, will add it to the title too.

P.S.2: while doing the diff between the makefiles of the 2 versions I found an space at the end of a description in v3, that's why I forcepushed to remove it.

Reply:
My opinion: adding the `help` target to go/v2's Makefile is fine, since it doesn't actually change any documented features/code, but changing `build` to `manager` is not because existing go/v2 projects won't be able to follow the same docs -> breaking change.

Reply:
@estroz that is why I left `manager: build` alias target with a help message that says that `manager` is deprecated.

Line 111: https://github.com/kubernetes-sigs/kubebuilder/pull/1981/files#diff-289e61cdb2aae0c8de547a662a2592227f5f77d7628850ffe609e922db923606R111

Reply:
It feels strange to scaffold a deprecated make target on `init`. As a user I could see being confused by usage despite the comments. How about adding `help` without aliasing `manager` as `build` for go/v2?

Reply:
The deprecation is just a note in the help message. Does it feel that strange?

Reply:
why not remove the deprecated note and keep is v3? shows that technically we would not introducing a breaking change at all.

Reply:
Changed the note to remove the deprecation part and just leave it as an alias for build to keep backwards compatibility.